c_level_db.go

```diff
@@ -303,55 +296,41 @@ func (itr cLevelDBIterator) Valid() bool {
 	return true
 }
 
-func (itr cLevelDBIterator) Key() ([]byte, error) {
-	if err := itr.assertNoError(); err != nil {
-		return nil, err
-	}
-	if err = itr.assertIsValid(); err != nil {
-		return nil, err
-	}
-	return itr.source.Key(), nil
+func (itr cLevelDBIterator) Key() []byte {
+	itr.assertNoError()
+	itr.assertIsValid()
+	return itr.source.Key()
 }
 
-func (itr cLevelDBIterator) Value() ([]byte, error) {
-	if err := itr.assertNoError(); err != nil {
-		return nil, err
-	}
-	if err = itr.assertIsValid(); err != nil {
-		return nil, err
-	}
-	return itr.source.Value(), nil
+func (itr cLevelDBIterator) Value() []byte {
+	itr.assertNoError()
+	itr.assertIsValid()
+	return itr.source.Value()
 }
 
-func (itr cLevelDBIterator) Next() error {
-	if err := itr.assertNoError(); err != nil {
-		return err
-	}
-	if err = itr.assertIsValid(); err != nil {
-		return err
-	}
+func (itr cLevelDBIterator) Next() {
+	itr.assertNoError()
+	itr.assertIsValid()
 	if itr.isReverse {
 		itr.source.Prev()
 	} else {
 		itr.source.Next()
 	}
-	return nil
 }
 
 func (itr cLevelDBIterator) Close() {
 	itr.source.Close()
 }
 
-func (itr cLevelDBIterator) assertNoError() error {
-	if err := itr.source.GetError(); err != nil {
-		return err
+func (itr cLevelDBIterator) assertNoError() {
+	err := itr.source.GetError()
+	if err != nil {
+		panic(err)
 	}
-	return nil
 }
 
-func (itr cLevelDBIterator) assertIsValid() error {
+func (itr cLevelDBIterator) assertIsValid() {
 	if !itr.Valid() {
-		return errors.New("cLevelDBIterator is invalid")
+		panic(errors.New("cLevelDBIterator is invalid"))
```

Comment:
```suggestion
		panic("cLevelDBIterator is invalid")
```

Reply:
ah sorry, didn't mean for you to review, still needed to do a few things